src/main/java/org/datadog/jmxfetch/util/JeeStatisticsAttributes.java

```diff
@@ -0,0 +1,328 @@
+            mhStatGetStatistic = maybeFindMethodHandleFor(classStat, "getStatistic", String.class);
+            classStatistic = maybeLookupClass("javax.management.j2ee.statistics.Statistic",
+                classLoader);
+            classCountStatistic =
+                maybeLookupClass("javax.management.j2ee.statistics.CountStatistic", classLoader);
+            classTimeStatistic =
+                maybeLookupClass("javax.management.j2ee.statistics.TimeStatistic", classLoader);
+            classRangeStatistic =
+                maybeLookupClass("javax.management.j2ee.statistics.RangeStatistic", classLoader);
+            classBoundaryStatistic =
+                maybeLookupClass("javax.management.j2ee.statistics.BoundaryStatistic", classLoader);
+            classBoundedRangeStatistic =
+                maybeLookupClass("javax.management.j2ee.statistics.BoundedRangeStatistic",
+                    classLoader);
+            methodCache = buildMethodCache();
+        }
+
+        private Map<Class<?>, Map<String, MethodHandle>> buildMethodCache() {
+            Map<Class<?>, Map<String, MethodHandle>> map = new HashMap<>();
+            if (classCountStatistic != null) {
+                map.put(classCountStatistic,
+                    buildMethodCacheFor(classCountStatistic, COUNT_ATTRIBUTES));
+            }
+            if (classTimeStatistic != null) {
+                map.put(classTimeStatistic,
+                    buildMethodCacheFor(classTimeStatistic, TIME_ATTRIBUTES));
+            }
+            if (classBoundaryStatistic != null) {
+                map.put(
+                    classBoundaryStatistic,
+                    buildMethodCacheFor(classBoundaryStatistic, BOUNDARY_ATTRIBUTES));
+            }
+            if (classRangeStatistic != null) {
+                map.put(classRangeStatistic,
+                    buildMethodCacheFor(classRangeStatistic, RANGE_ATTRIBUTES));
+            }
+            if (classBoundedRangeStatistic != null) {
+                map.put(
+                    classBoundedRangeStatistic,
+                    buildMethodCacheFor(classBoundedRangeStatistic, BOUNDED_RANGE_ATTRIBUTES));
+            }
+            return map;
+        }
+
+        private static MethodHandle maybeFindMethodHandleFor(
+            final Class<?> cls, final String name, final Class<?>... parameterTypes) {
+            if (cls == null) {
+                return null;
+            }
+            return AccessController.doPrivileged(
+                new PrivilegedAction<MethodHandle>() {
+                    @Override
+                    public MethodHandle run() {
+                        try {
+                            return MethodHandles.lookup()
+                                .unreflect(cls.getMethod(name, parameterTypes));
+                        } catch (Throwable t) {
+                            LOGGER.warn("Unable to find method {} for class {}: {}", name, cls,
```

Comment:
`maybeFindMethodHandleFor` suggests we expect sometimes not to find the method - in which case should this be logging as `debug` instead of `warn`?